x/tokenfactory/keeper/createdenom.go

```diff
@@ -3,61 +3,57 @@ package keeper
 import (
 	"fmt"
 
-	"cosmossdk.io/errors"
-
 	sdk "github.com/cosmos/cosmos-sdk/types"
 	banktypes "github.com/cosmos/cosmos-sdk/x/bank/types"
 
 	"github.com/neutron-org/neutron/x/tokenfactory/types"
 )
 
-// CreateDenom creates new denom with `factory/{creatorAddr}/{subdenom}` name.
-// Charges creatorAddr fee for creation
-func (k Keeper) CreateDenom(ctx sdk.Context, creatorAddr, subdenom string) (newTokenDenom string, err error) {
-	err = k.chargeFeeForDenomCreation(ctx, creatorAddr)
-	if err != nil {
-		return "", errors.Wrapf(types.ErrUnableToCharge, "denom fee collection error: %v", err)
-	}
-
+// ConvertToBaseToken converts a fee amount in a whitelisted fee token to the base fee token amount
+func (k Keeper) CreateDenom(ctx sdk.Context, creatorAddr string, subdenom string) (newTokenDenom string, err error) {
 	denom, err := k.validateCreateDenom(ctx, creatorAddr, subdenom)
 	if err != nil {
-		return "", errors.Wrapf(types.ErrInvalidDenom, "denom validation error: %v", err)
+		return "", err
 	}
 
-	err = k.createDenomAfterValidation(ctx, creatorAddr, denom)
+	err = k.chargeForCreateDenom(ctx, creatorAddr)
 	if err != nil {
-		return "", errors.Wrap(err, "create denom after validation error")
+		return "", err
 	}
 
-	return denom, nil
+	err = k.createDenomAfterValidation(ctx, creatorAddr, denom)
+	return denom, err
 }
 
 // Runs CreateDenom logic after the charge and all denom validation has been handled.
 // Made into a second function for genesis initialization.
-func (k Keeper) createDenomAfterValidation(ctx sdk.Context, creatorAddr, denom string) (err error) {
-	denomMetaData := banktypes.Metadata{
-		DenomUnits: []*banktypes.DenomUnit{{
-			Denom:    denom,
-			Exponent: 0,
-		}},
-		Base: denom,
-	}
+func (k Keeper) createDenomAfterValidation(ctx sdk.Context, creatorAddr string, denom string) (err error) {
+	_, exists := k.bankKeeper.GetDenomMetaData(ctx, denom)
```

Comment:
redundant validation, we already done it with `validateCreateDenom` method

Reply:
This is the same as in original `osmosis` module